### ISSUE TITLE
Django and mysqlclient

### ISSUE THREAD

**Issue by brett-matthews** [UnitedIncome/serverless-python-requirements#467]
Im attempting to use serverless-python-requirements + serverless-wsgi with Django using mysql database.

Ive read over the README numerous times but haven't been able to get this to work. 

Ive created a repo as an example. 
https://github.com/brett-matthews/serverless-django-wsgi-requirements-test

I would be happy to contribute to the README here if we could manage to solve this which may help other users in the future.

**Comment by miketheman** [comment]
@brett-matthews can you be more specific about what isn't working? Thanks for providing the repo link, but any error messages, console output are going to provide more context.

**Comment by miketheman** [comment]
The only error I've received when running `sls package` was a notice about the libmysql not being found, so I swapped the commented lines and was able to package.
https://github.com/brett-matthews/serverless-django-wsgi-requirements-test/blob/ccd5526a860da8097ed2bd9a585fa8e975f6eb17/serverless.yml#L10-L11

Are you getting a runtime error? Is this the same as #317 ?

**Comment by brett-matthews** [comment]
My bad I should of attached the Error @miketheman.. Added to original post.

Let me try again re:
https://github.com/brett-matthews/serverless-django-wsgi-requirements-test/blob/ccd5526a860da8097ed2bd9a585fa8e975f6eb17/serverless.yml#L10-L11

I left the commented out lines to demonstrate I tried both.. Or so I thought i did :)

**Comment by brett-matthews** [comment]
Still receive this error (from original post)

```
ImportError: libmysqlclient.so.18: cannot open shared object file: No such file or directory
```

Regardless of either of these options in `dockerExtraFiles`; tried --force on deploy too

```
      - /usr/lib64/mysql57/libmysqlclient.so.1020
      - /usr/lib64/mysql/libmysqlclient.so.18
```

**Comment by brett-matthews** [comment]
So I set these values in my serverless.yml file and now it is working!

```
  pythonRequirements:
    useDownloadCache: false
    useStaticCache: false
```

Im not exactly sure why, perhaps I had a cached build that was broken?

**Comment by littlebtc** [comment]
> So I set these values in my serverless.yml file and now it is working!
> 
> ```
>   pythonRequirements:
>     useDownloadCache: false
>     useStaticCache: false
> ```
> 
> Im not exactly sure why, perhaps I had a cached build that was broken?

I had ran into similar issues, it seems that the package inclusion/exclusion process will not re-run if you have static cache.

Remove the static cache or invalidate it by changing `requirements.txt` should re-package correctly.

As you had successfully built it, it should be safe to enable caches again.

**Comment by miketheman** [comment]
@brett-matthews Glad to hear it's working! I've tagged this to leave open for now.

I know this might be hard to confirm - in the README we note that you should try and confirm that your file was included in the zip package:
https://github.com/UnitedIncome/serverless-python-requirements/blob/0738379e04d43a120a800a66b40f97eb02a100d8/README.md#native-code-dependencies-during-runtime

Is that something you tried while debugging this?

Going to leave this open for others to see for now, at least until we learn more about the behaviors and consider what might be a good tradeoff between automatically clearing caches for extra files, or some other approach.

**Comment by brett-matthews** [comment]
As in looking to see what files were included in the Docker image?

https://github.com/brett-matthews/serverless-django-wsgi-requirements-test/blob/ccd5526a860da8097ed2bd9a585fa8e975f6eb17/serverless.yml#L10-L11

For someone not familiar with Docker such as myself this took a bit of research to try this file`- /usr/lib64/mysql/libmysqlclient.so.18` which was different to the tutorial. With the caching trouble I experienced it was even more confusing. 

Not sure on the direction you could take? Maybe a simple note in the Native Code Dependencies During Runtime section of the README to turn of caching off initially until happy with a working build?

**Comment by kahlil29** [comment]
@brett-matthews Thanks for opening this issue, providing so many details and leaving it open. Helped me after I hit this issue when trying to use a Python resizing Lambda with `Pillow-SIMD` dependency. 
Struggled for a while before coming across this issue and reading it helped my understanding of the issue and feature.

**Comment by zees-dev** [comment]
@brett-matthews Thanks from me too.
Was trying to run Django 3.1 serverlessly - hit roadblock once i found out i couldn't connect to remote MySQL DB due to a mysqlclient error (similar to yours).
I encountered the following error (after a long stack trace): `NameError: name '_mysql' is not defined`
Have successfully resolved the issue now (thanks to the fix you've provided) 👍

**Comment by zees-dev** [comment]
For anyone else who would like to get Django 3.1 (latest at this time) running on serverless infrastructure while also being able to connect to MySQL database; i have setup a repo for this:
https://github.com/zees-dev/serverless-django
